sympy/ntheory/factor_.py

```diff
@@ -1419,6 +1424,63 @@ def divisor_count(n, modulus=1):
     return Mul(*[v + 1 for k, v in factorint(n).items() if k > 1])
 
 
+def proper_divisors(n, generator=False):
+    """
+    Return all divisors of n except n, sorted by default.
+    If generator is ``True`` an unordered generator is returned.
+
+    Examples
+    ========
+
+    >>> from sympy import proper_divisors, proper_divisor_count
+    >>> proper_divisors(24)
+    [1, 2, 3, 4, 6, 8, 12]
+    >>> proper_divisor_count(24)
+    7
+    >>> list(proper_divisors(120, generator=True))
+    [1, 2, 4, 8, 3, 6, 12, 24, 5, 10, 20, 40, 15, 30, 60]
+
+    See Also
+    ========
+
+    factorint, divisors, proper_divisor_count
+
+    """
+    n = as_int(abs(n))
```

Comment:
You are duplicating code here, too...so how about giving divisors a 3rd argument `proper=False` and then your `proper_divisors` is just a thin wrapper to `return divisors(n, generator, proper=True)`.

Reply:
@smichr Done, thanks! Please check it again